### ISSUE TITLE
Build dependencies fail if also declared as non-build dependencies.

### ISSUE THREAD

**Issue by dwrensha** [rust-lang/cargo#876]
```
#Cargo.toml

[package]

name = "foo"
version = "0.1.0"
build = "build.rs"
authors = ["David Renshaw"]

[lib]
name = "foo"
path = "lib.rs"


[build-dependencies.splay]
git = "https://github.com/alexcrichton/splay-rs.git"

[dependencies.splay]
git = "https://github.com/alexcrichton/splay-rs.git"
```

```
// build.rs

extern crate splay;

fn main() {}
```

```
// lib.rs

#![crate_name="foo"]
#![crate_type="lib"]

extern crate splay;
```

```
$ cargo build
   Compiling foo v0.1.0 (file:///Users/dwrensha/Desktop/test-cargo)
   Compiling splay v0.1.0 (https://github.com/alexcrichton/splay-rs.git#3b1e342a)
build.rs:3:1: 3:20 error: can't find crate for `splay`
build.rs:3 extern crate splay;
           ^~~~~~~~~~~~~~~~~~~
```

It succeeds if I comment out the `[dependencies.splay]` part.

**Comment by blaenk** [comment]
Yeah ran into this myself.

**Comment by andrewtj** [comment]
Is a fix for this likely to land soon?

**Comment by jakerr** [comment]
I attempted a fix but don't understand the code well enough to know if what I perceived as the possible cause actually was: https://github.com/rust-lang/cargo/commit/7f7dca39fdc7c5a1bb0e5d8039228f37e4576e3c

It seemed to work for a simple test case but caused weird missing rlibs and such when used in a real project. Comments on what I tried on the commit.

**Comment by mohtar** [comment]
Extended jakerr's fix to the job dependency queuing logic: 82e6e809d05ee65cea8571fe869495cf273e4ab9
